src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/ConsoleLifetimeExitTests.cs

```diff
@@ -69,6 +89,69 @@ await Host.CreateDefaultBuilder()
         [DllImport("libc", SetLastError = true)]
         private static extern int kill(int pid, int sig);
 
+
+        private const int CTRL_C_EVENT = 0;
+        private const int CTRL_BREAK_EVENT = 1;
+        private const int CTRL_CLOSE_EVENT = 2;
+        private const int CTRL_LOGOFF_EVENT = 5;
+        private const int CTRL_SHUTDOWN_EVENT = 6;
+
+        private static unsafe void SimulatePosixSignalWindows(string pipeHandleAsString)
+        {
+            try
+            {
+                using var readPipe = new AnonymousPipeClientStream(PipeDirection.In, pipeHandleAsString);
+
+                int signal = (int)readPipe.ReadByte();
+
+                int ctrlType = (int)signal switch
+                {
+                    SIGINT => CTRL_C_EVENT,
+                    SIGQUIT => CTRL_BREAK_EVENT,
+                    SIGTERM => CTRL_SHUTDOWN_EVENT,
+                    _ => throw new ArgumentOutOfRangeException(nameof(signal), "Unsupported signal")
+                };
+
+#if NETFRAMEWORK
+            if (ctrlType == CTRL_C_EVENT || ctrlType == CTRL_BREAK_EVENT)            
+            {
+                var handlerMethod = Type.GetType("System.Console, mscorlib")?.GetMethod("BreakEvent", BindingFlags.NonPublic | BindingFlags.Static);
+                Assert.NotNull(handlerMethod);
+                handlerMethod.Invoke(null, [ctrlType]);
+            }
+            else // CTRL_SHUTDOWN_EVENT
+            {
+                var handlerField = typeof(AppDomain).GetField("_processExit", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
This doesn't actually seem to be invoked on .NETFramework during a shutdown.  AFAICT a .NETFramework application still doesn't get **any** signal on Docker shutdown.